scaaml/io/resume_kti.py

```diff
@@ -165,15 +237,18 @@ def __next__(self):
         """
         if self._index % self._shard_length == 0:
             self._save_progress()
-        if self._index >= len(self._keys):
+        if self._index >= get_any_value_len(self._parameters):
             raise StopIteration
         self._index += 1
-        return self._keys[self._index - 1], self._texts[self._index - 1]
+        return {
```

Comment:
Same here, I think a namedtuple would look nicer. You can always turn the namedtuple into a dict using the `._asdict()` method or get a tuple with the field names using `._fields` property